### PR TITLE
Modernize gpg related formulae

### PR DESCRIPTION
**Pull request by xu-cheng**
Basically, I create an unified  `depends_on :gpg` DSL.

There's another gpg related formula `signing-party`. It's much more complicated, so I will revisit that formula in an separate PR.

### CODE REVIEW

File: Library/Formula/keybase.rb [hunk 1]
```diff
@@ -24,12 +16,6 @@ def install
     EOS
   end
 
-  def caveats;<<-EOS.undent if which("gpg2") && !which("gpg")
-      Run below command if you use gpg2 as keybase's backend
-        keybase config gpg gpg2
-    EOS
-  end
-
```
Comment (xu-cheng) [inline Library/Formula/keybase.rb]: Removed as the latest version keybase can handle this automatically.

File: Library/Formula/pius.rb [hunk 2]
```diff
@@ -1,34 +1,26 @@
...
-
-  def install
-    # Replace hardcoded gpg path: https://sourceforge.net/p/pgpius/bugs/12/
-    inreplace 'pius', '/usr/bin/gpg', HOMEBREW_PREFIX/'bin/gpg'
-    bin.install 'pius'
-    bin.install 'pius-keyring-mgr'
-    bin.install 'pius-party-worksheet'
+  test do
+    system "#{bin}/pius", "--version"
   end
```
Comment (xu-cheng) [inline Library/Formula/pius.rb]: I cannot find a better test without interfering users' gpg keyring.

File: Library/Formula/pius.rb [hunk 3]
```diff
@@ -1,34 +1,26 @@
...
+    bin.install "pius-party-worksheet"
+  end
 
-    Or you can use one of several different
-    prepackaged installers that are available.
+  def caveats; <<-EOS.undent
+      The gpg path is hardcoded in pius. You can specific your own gpg path
+      by add fellow code into ~/.pius file:
+        gpg-path=/path/to/gpg
     EOS
```
Comment (xu-cheng) [inline Library/Formula/pius.rb]: Caveats come from https://sourceforge.net/p/pgpius/bugs/12/.

File: Library/Formula/pius.rb (hunk 2 above)
Comment (mikemcquaid) [inline Library/Formula/pius.rb]: `HOME` is overriden in the tests if that helps?

**Comment by mikemcquaid** [comment]
Otherwise looks great! :+1:

File: Library/Formula/pius.rb (hunk 2 above)
Comment (xu-cheng) [inline Library/Formula/pius.rb]: Will look it when I have free time. (Such a busy week for a PhD student. :smile: )

Comment (mikemcquaid) [inline Library/Formula/pius.rb]: Thanks dude!

Comment (xu-cheng) [inline Library/Formula/pius.rb]: Silly me, this is a simple test I missed.

**Comment by xu-cheng** [comment]
This PR is ready to be merged.

**Pull request closed without merging**